File: SNS.Web/src/app/support/support-tickets/contracts/support-ticket-summary.dto.ts
import { SupportTicketCategory } from '../../enums/support-ticket-category.enum';
import { TicketPriority } from '../../enums/ticket-priority.enum';
import { TicketStatus } from '../../enums/ticket-status.enum';

export interface SupportTicketSummaryDto {
    id: string;
    userId: string;
    assignedAgentId: string | null;
    title: string;
    category: SupportTicketCategory;
    priority: TicketPriority;
    status: TicketStatus;
    messagesCount: number;
    createdAt: string;
    updatedAt: string;
}

File: backend/SNS.Application/Support/SupportTickets/Contracts/SupportTicketSummaryDto.cs
using SNS.Domain.Support.Enums;

namespace SNS.Application.Support.SupportTickets.Contracts;

public sealed record SupportTicketSummaryDto(
    Guid Id,
    Guid UserId,
    Guid? AssignedAgentId,
    string Title,
    SupportTeckitCategory Category,
    TicketPriority Priority,
    TicketStatus Status,
    int MessagesCount,
    DateTime CreatedAt,
    DateTime UpdatedAt);
